### PR TITLE
Fix #424 Add optimization step to set parameters and variables

### LINKED ISSUE #424: [Feature]: Support set parameter according to scenario
### Describe your use case

OceanBase has provided several parameter templates for common scenarios, set the parameters according to the templates will significantly improve the performance of OceanBase

### Describe the solution you'd like

add one step after create cluster or tenant to set the parameters

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chris-sun-star**
## Summary

1. add scenario configuration in obcluster and obtenant 
2. add optimization step to set parameters and variables

### CODE REVIEW

File: internal/resource/utils/optimize.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+You can use this software according to the terms and conditions of the Mulan PSL v2.
+You may obtain a copy of Mulan PSL v2 at:
+         http://license.coscl.org.cn/MulanPSL2
+THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+See the Mulan PSL v2 for more details.
+*/
+
+package utils
```
Comment (powerfooI) [inline internal/resource/utils/optimize.go:13]: What's the usage of this empty file?

File: internal/resource/obtenant/obtenant_task.go [hunk 2]
```diff
@@ -509,6 +515,44 @@ func CheckAndApplyLocality(m *OBTenantManager) tasktypes.TaskError {
 	return nil
 }
 
+func OptimizeTenantByScenario(m *OBTenantManager) tasktypes.TaskError {
+	// TODO: implement this
```
Comment (powerfooI) [inline internal/resource/obtenant/obtenant_task.go:519]: This `TODO` seems to have been done.

File: internal/cmds/helper/optimize.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	"os"
+
+	"github.com/spf13/cobra"
+
+	cmdconst "github.com/oceanbase/ob-operator/internal/const/cmd"
+	"github.com/oceanbase/ob-operator/pkg/helper/model"
+)
+
+const (
+	OptimizedParameterConfigFile = "/home/admin/oceanbase/etc/default_parameter.json"
```
Comment (powerfooI) [inline internal/cmds/helper/optimize.go:27]: How would we put the two files on desired path?

File: api/v1alpha1/obtenant_types.go [hunk 4]
```diff
@@ -50,6 +50,7 @@ type OBTenantSpec struct {
 	TenantRole  apitypes.TenantRole `json:"tenantRole,omitempty"`
 	Source      *TenantSourceSpec   `json:"source,omitempty"`
 	Credentials TenantCredentials   `json:"credentials,omitempty"`
+	Scenario    string              `json:"scenario,omitempty"`
```
Comment (powerfooI) [inline api/v1alpha1/obtenant_types.go:53]: Why does not tenant need default optimization scenario?

Comment (chris-sun-star) [reply api/v1alpha1/obtenant_types.go:53]: tenant will use the scenario of the obcluster it belongs to if it's not set, this logic is implemented in the tenant's webhook

File: internal/cmds/helper/optimize.go (hunk 3 above)
Comment (chris-sun-star) [reply internal/cmds/helper/optimize.go:27]: These two files are provided by OceanBase package

Comment (chris-sun-star) [reply internal/cmds/helper/optimize.go:27]: It's already in the image

File: internal/resource/utils/optimize.go (hunk 1 above)
Comment (chris-sun-star) [reply internal/resource/utils/optimize.go:13]: OK, I will delete it

File: internal/resource/obtenant/obtenant_task.go (hunk 2 above)
Comment (chris-sun-star) [reply internal/resource/obtenant/obtenant_task.go:519]: OK, I will delete it

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**